### PR TITLE
ujson Py2 compat removal

### PR DESCRIPTION
**Pull request by WillAyd**
There was a header file in the ujson source specifically meant to bridge compatibility between Py2 and Py3. With the Py2 support drop this could be removed and the source code simplified.

Most of these changes are simply verbatim, though there are probably a few follow ups on context to simplify further that could be made

### CODE REVIEW

File: pandas/_libs/src/ujson/python/JSONtoObj.c [hunk 1]
```diff
@@ -35,14 +35,12 @@ Numeric decoder derived from from TCL library
...
 
-// "py_defines.h" needs to be included first to
-// avoid compilation errors, but it does violate
-// styleguide checks with regards to include order.
-#include "py_defines.h"
 #define PY_ARRAY_UNIQUE_SYMBOL UJSON_NUMPY
 #define NO_IMPORT_ARRAY
-#include <numpy/arrayobject.h>  // NOLINT(build/include_order)
-#include <ultrajson.h>          // NOLINT(build/include_order)
+#define PY_SSIZE_T_CLEAN
```
Comment (WillAyd) [inline pandas/_libs/src/ujson/python/JSONtoObj.c]: This definition is suggested by the Python developers:

https://docs.python.org/3/c-api/intro.html#include-files

I noticed we haven't used anywhere else so could probably be added as a follow up in other places

File: pandas/_libs/src/ujson/python/objToJSON.c [hunk 2]
```diff
@@ -36,19 +36,16 @@ Numeric decoder derived from from TCL library
...
-// "py_defines.h" needs to be included first to
-// avoid compilation errors, but it does violate
-// styleguide checks with regards to include order.
-#include "py_defines.h"           // NOLINT(build/include_order)
-#include <math.h>                 // NOLINT(build/include_order)
-#include <numpy/arrayobject.h>    // NOLINT(build/include_order)
-#include <numpy/arrayscalars.h>   // NOLINT(build/include_order)
-#include <numpy/ndarraytypes.h>   // NOLINT(build/include_order)
-#include <numpy/npy_math.h>       // NOLINT(build/include_order)
-#include <stdio.h>                // NOLINT(build/include_order)
```
Comment (WillAyd) [inline pandas/_libs/src/ujson/python/objToJSON.c]: `#include <Python.h>` implies this so no need to state again

File: pandas/_libs/src/ujson/python/JSONtoObj.c [hunk 3]
```diff
@@ -530,7 +528,7 @@ PyObject *JSONToObj(PyObject *self, PyObject *args, PyObject *kwargs) {
         decoder->preciseFloat = 1;
     }
 
-    if (PyString_Check(arg)) {
+    if (PyBytes_Check(arg)) {
```
Comment (jreback) [inline pandas/_libs/src/ujson/python/JSONtoObj.c]: so if this is correct then the error message is wrong? should be bytes or unicode? is there something that hits this in testing?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/src/ujson/python/JSONtoObj.c (hunk 3 above)
Comment (WillAyd) [reply pandas/_libs/src/ujson/python/JSONtoObj.c]: Updated the error message and added a test to test_ujson. FWIW I don't think anything user-facing hits this as doing:

```python
pd.read_json(None)
```

Would yield:

```python-traceback
>>> pd.read_json(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/williamayd/clones/pandas/pandas/io/json/json.py", line 437, in read_json
    path_or_buf, encoding=encoding, compression=compression,
  File "/Users/williamayd/clones/pandas/pandas/io/common.py", line 207, in get_filepath_or_buffer
    raise ValueError(msg.format(_type=type(filepath_or_buffer)))
ValueError: Invalid file path or buffer object type: <class 'NoneType'>
```

You could hit it directly internally if you did something like:

```python
import pandas._libs.json as ujson
ujson.loads(None)
```

which now yields:

```python-traceback
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Expected 'str' or 'bytes'
```

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @WillAyd

**Pull request merged**